### PR TITLE
meta: added million.js compiler (react) and updated packages

### PR DESCRIPTION
**Pull request by ovflowd**
This PR adds the Million.js Compiler, which supposedly drastically reduces React's re-rendering times and complexity by "certain" bypasses of its reconciliation algorithm and another sort of magic.

This PR also updates all our packages to latest ✅ versions.

### CODE REVIEW

File: next.config.mjs [hunk 1]
```diff
@@ -127,10 +133,13 @@ const sentryConfig = {
 // Next.js Configuration with `next.intl` enabled
 const nextWithIntl = withNextIntl('./i18n.tsx')(nextConfig);
 
+// Enables Million.js React Compiler for Next.js
+const nextWithMillion = million.next(nextWithIntl, { auto: true, rsc: true });
```
Comment (AugustinMauroy) [inline next.config.mjs:137]:
```suggestion
const nextWithMillion = million.next(nextWithIntl, {
  // support react server components
  auto: { rsc: true },
  // We don't want to send telemetry data to Million
  telemetry: false,
});
```

[docs ref for rsc](https://million.dev/docs/install#add-the-compiler-to-your-application)
[docs ref for telemetry](https://million.dev/telemetry#how-do-i-opt-out)

**Review by AugustinMauroy — COMMENTED** [review]
Also you should use the [`For` component](https://million.dev/docs/manual-mode/for).
it would certainly avoid this
Capture d’écran 2024-03-17 à 14 36 26

Other point, you should add million in docs.

**Comment by AugustinMauroy** [comment]
also strange behaviour search box is accessible by `cmd + k` but not on click

**Comment by ovflowd** [comment]
Well if this addition is not a drop-in placement then I don't think we should adopt the Million.js at the moment.

I don't think that wrapping each iteration with this For component and other stuff is worth it.

I don't want us to get dependent on yet another ecosystem.

Im closing this

**Comment by ovflowd** [comment]
ALSO, I assume this is the issue with Million.js duplicating and crashing: https://github.com/aidenybai/million/issues/991

**Pull request closed without merging**